### PR TITLE
Warn if call type job is used for constant function

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean.young@monax.io>

### CODE REVIEW

File: execution/evm/abi/abi.go [hunk 1]
```diff
@@ -1165,7 +1166,7 @@ func (abiSpec *AbiSpec) UnpackWithID(data []byte, args ...interface{}) error {
 	})
 }
 
-func (abiSpec *AbiSpec) Pack(fname string, args ...interface{}) ([]byte, error) {
+func (abiSpec *AbiSpec) Pack(fname string, args ...interface{}) ([]byte, bool, error) {
```
Comment (silasdavis) [inline execution/evm/abi/abi.go]: Can we have a doc comment on this, particularly for what the return values are

File: execution/evm/abi/core.go [hunk 2]
```diff
@@ -27,10 +27,10 @@ func init() {
 
 var RevertAbi *AbiSpec
 
-func ReadAbiFormulateCallFile(abiLocation, binPath, funcName string, args []string) ([]byte, error) {
+func ReadAbiFormulateCallFile(abiLocation, binPath, funcName string, args []string) ([]byte, bool, error) {
```
Comment (silasdavis) [inline execution/evm/abi/core.go]: doc comment

File: execution/evm/abi/core.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ func ReadAbiFormulateCallFile(abiLocation, binPath, funcName string, args []stri
 	return Packer(abiSpecBytes, funcName, args...)
 }
 
-func ReadAbiFormulateCall(abiSpecBytes []byte, funcName string, args []string) ([]byte, error) {
+func ReadAbiFormulateCall(abiSpecBytes []byte, funcName string, args []string) ([]byte, bool, error) {
```
Comment (silasdavis) [inline execution/evm/abi/core.go]: doc comment

File: execution/evm/abi/core.go [hunk 4]
```diff
@@ -63,30 +63,30 @@ func ReadAndDecodeContractReturn(abiLocation, binPath, funcName string, resultRa
 }
 
 //Convenience Packing Functions
-func Packer(abiData, funcName string, args ...string) ([]byte, error) {
+func Packer(abiData, funcName string, args ...string) ([]byte, bool, error) {
```
Comment (silasdavis) [inline execution/evm/abi/core.go]: doc comment

File: execution/evm/abi/abi.go (hunk 1 above)
Comment (seanyoung) [reply execution/evm/abi/abi.go]: Add godoc

File: execution/evm/abi/core.go (hunk 2 above)
Comment (seanyoung) [reply execution/evm/abi/core.go]: Add godoc and give function sensible name

File: execution/evm/abi/core.go (hunk 3 above)
Comment (seanyoung) [reply execution/evm/abi/core.go]: Add godoc and give function sensible name

File: execution/evm/abi/core.go (hunk 4 above)
Comment (seanyoung) [reply execution/evm/abi/core.go]: function gone

**Pull request merged**